--- a/drivers/staging/media/imx/imx8-media-dev.c
+++ b/drivers/staging/media/imx/imx8-media-dev.c
@@ -533,13 +533,18 @@
 				if (ret)
 					return ret;
 
+				v4l2_info(&mxc_md->v4l2_dev, "Linking %s:%d to %s:%d\n",
+					  source->name, source_pad + j, sink->name, sink_pad + j);
+
 				/* Notify MIPI subdev entity */
 				ret = media_entity_call(sink, link_setup,
 							&sink->pads[sink_pad + j],
 							&source->pads[source_pad + j],
 							0);
-				if (ret)
+				if (ret && ret != -ENOIOCTLCMD) {
+					v4l2_err(&mxc_md->v4l2_dev, "CSI link_setup failed (error %d)\n", ret);
 					return ret;
+				}
 
 				/* Notify MIPI sensor subdev entity */
 				ret = media_entity_call(source, link_setup,
@@ -546,8 +551,10 @@
 				ret = media_entity_call(source, link_setup,
 							&sink->pads[sink_pad + j],
 							0);
-				if (ret)
+				if (ret && ret != -ENOIOCTLCMD) {
+					v4l2_err(&mxc_md->v4l2_dev, "Sensor link_setup failed (error %d)\n", ret);
 					return ret;
+				}
 			}
 			v4l2_info(&mxc_md->v4l2_dev,
 				  "created link [%s] => [%s]\n",
